lib/util/util.HashMessageProvider.js

```diff
@@ -0,0 +1,47 @@
+this.util = this.util || {};
+
+util.HashMessageProvider = ( function() {
```

Comment:
Here you get me with naming... Correct me if I am wrong my "hash" does not necessarily mean "hash map" or "hash table".